spec.html

```diff
@@ -5713,38 +5713,20 @@ <h1>CreateIterResultObject ( _value_, _done_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-createlistiteratorRecord" oldids="sec-createlistiterator" aoid="CreateListIteratorRecord">
+    <emu-clause id="sec-createlistiteratorRecord" oldids="sec-createlistiterator,sec-listiteratornext-functions,sec-listiterator-next" aoid="CreateListIteratorRecord">
       <h1>CreateListIteratorRecord ( _list_ )</h1>
       <p>The abstract operation CreateListIteratorRecord takes argument _list_. It creates an Iterator (<emu-xref href="#sec-iterator-interface"></emu-xref>) object record whose next method returns the successive elements of _list_. It performs the following steps when called:</p>
       <emu-alg>
-        1. Let _iterator_ be ! OrdinaryObjectCreate(%IteratorPrototype%, &laquo; [[IteratedList]], [[ListNextIndex]] &raquo;).
-        1. Set _iterator_.[[IteratedList]] to _list_.
-        1. Set _iterator_.[[ListNextIndex]] to 0.
-        1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-listiteratornext-functions" title></emu-xref>.
-        1. Let _next_ be ! CreateBuiltinFunction(_steps_, &laquo; &raquo;).
-        1. Return Record { [[Iterator]]: _iterator_, [[NextMethod]]: _next_, [[Done]]: *false* }.
+        1. Let _closure_ be a new Abstract Closure with no parameters that captures _list_ and performs the following steps when called:
+          1. For each element _E_ of _list_, do
+            1. Perform ? Yield(_E_).
+          1. Return *undefined*.
+        1. Let _iterator_ be ! CreateIteratorFromClosure(_closure_, ~empty~, %IteratorPrototype%).
+        1. Return Record { [[Iterator]]: _iterator_, [[NextMethod]]: %Generator.prototype.next%, [[Done]]: *false* }.
```

Comment:
The generator intrinsics were renamed in <https://github.com/tc39/ecma262/pull/2065>:

```suggestion
        1. Return Record { [[Iterator]]: _iterator_, [[NextMethod]]: %GeneratorFunction.prototype.prototype.next%, [[Done]]: *false* }.
```